habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PublishToPyPiRepoMojo.java

```diff
@@ -308,14 +308,13 @@ protected Retryer<Boolean> getRetryer() {
     }
 
     String getRepositoryUrl(boolean publishToDev) {
-        String repoUrl;
+        String repoUrl = addTrailingSlash(pypiRepoUrl);
         if (publishToDev) {
-            repoUrl = addTrailingSlash(devRepositoryUrl) + devRepositoryUrlUploadSuffix;
-        } else {
-            repoUrl = addTrailingSlash(pypiRepoUrl) + pypiUploadSuffix;
+            repoUrl = addTrailingSlash(devRepositoryUrl) + addTrailingSlash(devRepositoryUrlUploadSuffix);
+        } else if(!pypiUploadSuffix.isEmpty()) {
```

Comment:
a: Could use StringUtils to prevent npe in case of the upload suffix being set to null

Reply:
Changed